include/aws/io/channel_bootstrap.h

```diff
@@ -294,6 +294,14 @@ AWS_IO_API int aws_server_bootstrap_set_alpn_callback(
 AWS_IO_API struct aws_socket *aws_server_bootstrap_new_socket_listener(
     const struct aws_server_socket_channel_bootstrap_options *bootstrap_options);
 
+/**
+ * Get local bound port for network sockets.
+ * @retval -1 on errors
+ */
+AWS_IO_API int aws_server_bootstrap_get_bound_port_of_socket_listener(
```

Comment:
This function seems redundant given the other one.

Reply:
It depends on strategy.

For strict consistency, the best case is reading the actual port directly in aws_socket_bind() and updating the field of struct aws_socket_endpoint *local_endpoint. Unfortunately local_endpoint argument is defined as a pointer to const struct.
Removing a const specifier means drop backward compatibility. Also changes in aws_socket_bind() can affect any existing applications which do not expect the port in local_endpoint will be changed by aws_socket_bind(). Additional third output argument means changing function signature and updating all aws_socket_bind() callers, in all existing applications.

For strict backwards compatibility best is to implement an isolated chain of functions to provide new functions and keep unmodified all existing applications and library functions used by it. Only new applications which use new functions can cause problems even when bugs exist in new code.

Because I'm newbie in aws-crt-java and it's 18 submodules, I selected the second approach.

Reply:
It is wrapper function for aws_event_stream_rpc_server_listener_get_bound_port of aws-c-event-stream. Please take a look https://github.com/awslabs/aws-c-event-stream/pull/78 

Unfortunately  event_stream_rpc_server.c does not use aws_socket_ symbols directly.
I just follow existing path of calls.